Compile time error when using new `bedjet` component `no member named 'set_time_id'`

### The problem

The ESPHome docs page for the BedJet component point out that there is a way to set the clock on the bed jet. 

https://github.com/esphome/esphome-docs/blame/56ff044ab3f350a29c262195ea49cd59b2015ed8/components/climate/bedjet.rst#L42

However, when attempting to use the component, this error is emitted during compile/build:

```shell
src/main.cpp: In function 'void setup()':
src/main.cpp:507:16: error: 'class esphome::bedjet::Bedjet' has no member named 'set_time_id'
   bedjet_f00b->set_time_id(sntp_time);
```

### Which version of ESPHome has the issue?

Version: 2022.5.0

### What type of installation are you using?

pip

### Which version of Home Assistant has the issue?

_No response_

### What platform are you using?

ESP32

### Board

_No response_

### Component causing the issue

BedJet

### Example YAML snippet

```yaml
# See: https://esphome.io/components/climate/index.html
climate:
  - platform: bedjet
    id: bedjet_f00b
    name: "BedJet"
    ble_client_id: ble_bedjet_f00b
    icon: "mdi:turbine"
    # Sets the bedjet clock
    # Getting compile time errors: src/main.cpp:507:16: error: 'class esphome::bedjet::Bedjet' has no member named 'set_time_id'
    time_id: sntp_time
    visual:
      # Not clear to me what unit; i am assuming celsius
      ##
      # Cooling does not let you go below 19
      min_temperature: 19

      # On the "extra turbo heat" setting, you can go as high as 43c
      max_temperature: 43
      # Bedjet app does 1c increments
      temperature_step: 1
```


### Anything in the logs that might be useful for us?

_No response_

### Additional information

It looks like the header file does declare the function, so i'm not sure why `USE_TIME` is not set for me.
https://github.com/esphome/esphome/blob/dev/esphome/components/bedjet/bedjet.h#L38

--- a/esphome/components/bedjet/bedjet.h
+++ b/esphome/components/bedjet/bedjet.h
@@ -4,6 +4,7 @@
 #include "esphome/components/esp32_ble_tracker/esp32_ble_tracker.h"
 #include "esphome/components/climate/climate.h"
 #include "esphome/core/component.h"
+#include "esphome/core/defines.h"
 #include "esphome/core/hal.h"
 #include "bedjet_base.h"
 

